drivers/espi/espi_mchp_xec.c

```diff
@@ -580,30 +581,42 @@ static void espi_vwire_chanel_isr(struct device *dev)
 	espi_send_callbacks(&data->callbacks, dev, evt);
 }
 
+#ifdef CONFIG_ESPI_OOB_CHANNEL
 static void espi_oob_down_isr(struct device *dev)
 {
 	u32_t status;
 
+	LOG_DBG("%s\n", __func__);
+
 	status = ESPI_OOB_REGS->RX_STS;
 	if (status & MCHP_ESPI_OOB_RX_STS_DONE) {
 		ESPI_OOB_REGS->RX_IEN = ~MCHP_ESPI_OOB_RX_IEN;
 	}
-
-	GIRQ19_REGS->SRC = MCHP_ESPI_OOB_DN_GIRQ_VAL;
 }
 
 static void espi_oob_up_isr(struct device *dev)
 {
 	u32_t status;
 
+	LOG_DBG("%s\n", __func__);
+
 	status = ESPI_OOB_REGS->TX_STS;
 	if (status & MCHP_ESPI_OOB_TX_STS_DONE) {
-		ESPI_OOB_REGS->TX_IEN = ~MCHP_ESPI_OOB_TX_IEN_DONE;
+		ESPI_OOB_REGS->TX_IEN |= MCHP_ESPI_OOB_TX_IEN_DONE;
```

Comment:
@scottwcpg I'm clearing GIRQ19 bits inside espi_xec_bus_isr just before calling espi_oob_up_isr. 

Could you elaborate in the first issue regarding OOB_TX done.  I'm clearing ESPI_OOB_REGS->TX_STS by writting 1, before mask used was wrong.

In separate case, I'm checking channel status change bit and re-enabling interrupts whenever channel gets re-enabled.

Reply:
The lowest level interrupt status must be cleared first before clearing GIRQ19 bits.  GIRQ19 OOB status should be cleared after OOB TX/RX status register sticky bits are cleared.  When I look at espi_xec_bus_isr it reads the result register and then writes that value to GIRQ19 status register to clear all active status.  Since the individual sources of the status bits have not been serviced and cleared those status bits will go active again on the next AHB clock.  We implement an aggregated handler by reading result looping on the result bits calling a handler for each active bit, when the handler returns after clearing the low level status the loop code then clears the corresponding bit in the GIRQ Source register before looping around.

Reply:
So basically moving REG32(MCHP_GIRQ_SRC_ADDR(config->bus_girq_id)) = girq_result; to the end of the function, correct?

Reply:
Yes, that's the easiest change.  The other method would be slower as means clearing each active bit after the if (entry.isr != NULL) { call function }; clear the GIRQ bit; ...

Reply:
Fixed